pydarn/io/borealis.py

```diff
@@ -0,0 +1,1298 @@
+                        v['sqn_timestamps'][0]).day),
+                    'time.hr': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).hour),
+                    'time.mt': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).minute),
+                    'time.sc': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).second),
+                    'time.us': np.int32(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).microsecond),
+                    'txpow': np.int16(-1),
+                    'nave': np.int16(v['num_sequences']),
+                    'atten': np.int16(0),
+                    'lagfr': np.int16(v['first_range_rtt']),
+                    'smsep': np.int16(1e6/v['rx_sample_rate']),
+                    'ercod': np.int16(0),
+                    'stat.agc': np.int16(0),
+                    'stat.lopwr': np.int16(0),
+                    'noise.search': np.float32(v['noise_at_freq'][0]),
+                    'noise.mean': np.float32(0),
+                    'channel': np.int16(slice_id),
+                    'bmnum': np.int16(beam),
+                    'bmazm': np.float32(v['beam_azms'][beam_index]),
+                    'scan': np.int16(v['scan_start_marker']),
+                    'offset': np.int16(0),
+                    'rxrise': np.int16(0),
+                    'intt.sc': np.int16(math.floor(v['int_time'])),
+                    'intt.us': np.int32(math.fmod(v['int_time'], 1.0) * 1e6),
+                    'txpl': np.int16(v['tx_pulse_len']),
+                    'mpinc': np.int16(v['tau_spacing']),
+                    'mppul': np.int16(len(v['pulses'])),
+                    # an alternate lag-zero will be given.
+                    'mplgs': np.int16(v['lags'].shape[0] - 1),
+                    'nrang': np.int16(v['num_ranges']),
+                    'frang': np.int16(v['first_range']),
+                    'rsep': np.int16(v['range_sep']),
+                    'xcf': np.int16('intf' in v['antenna_arrays_order']),
+                    'tfreq': np.int16(v['freq']),
+                    'mxpwr': np.int32(-1),
+                    'lvmax': np.int32(20000),
+                    'iqdata.revision.major': np.int32(1),
+                    'iqdata.revision.minor': np.int32(0),
+                    'combf': 'Converted from Borealis file: ' + \
+                        self.filename  + ' record ' + k + \
+                        ' ; Number of beams in record: ' + \
+                        str(len(v['beam_nums'])) + ' ; ' + \
+                        v['experiment_comment'] + ' ; ' + v['slice_comment'],
+                    'seqnum': np.int32(v['num_sequences']),
+                    'chnnum': np.int32(v['antenna_arrays_order'].shape[0]),
+                    'smpnum': np.int32(v['num_samps']),
+                    # NOTE: The following is a hack. This is currently how 
```

Comment:
documentation could be improved to indicate that it's used as a number of samples to first range. Otherwise not much can be done because it's written in ROS.